presto-spi/src/main/java/com/facebook/presto/spi/block/ArrayAllocator.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.spi.block;
+
+import com.facebook.presto.spi.api.Experimental;
+
+@Experimental
+public interface ArrayAllocator
+{
+    int[] getIntArray(int positionCount);
+
+    void returnArray(int[] array);
```

Comment:
I feel returnArray() is not quite self-explanatory. Can we rename them as checkoutIntArray and checkinIntArray()?

Reply:
Per offline conversation, I will call it `borrowIntArray` and `returnArray`